src/containers/CustomRuns/CustomRuns.jsx

```diff
@@ -294,6 +317,30 @@ function CustomRuns() {
     ];
   }
 
+  const toolbarButtons = isReadOnly
+    ? []
+    : [
+        {
+          onClick: () => {
+            let queryString;
+            if (namespace !== ALL_NAMESPACES) {
+              queryString = new URLSearchParams({
+                ...(namespace !== ALL_NAMESPACES && { namespace })
+              }).toString();
+            }
+            navigate(
+              // currently default is yaml mode
+              urls.customRuns.create() + (queryString ? `?${queryString}&mode=yaml` : '?mode=yaml')
+            );
```

Comment:
For now we'll always have at least 1 query param (`mode=yaml`) so we can simplify this and always construct the query string using URLSearchParams:

```suggestion
            const queryString = new URLSearchParams({
                ...(namespace !== ALL_NAMESPACES && { namespace }),
                // currently default is yaml mode
                mode: 'yaml'
              }).toString();
            }
            navigate(urls.customRuns.create() + `?${queryString}`);
```